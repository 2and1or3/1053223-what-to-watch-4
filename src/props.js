import PropTypes from "prop-types";

const filmProp = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  preview: PropTypes.string.isRequired,
  background: PropTypes.string.isRequired,
  cover: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  release: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  voiceCount: PropTypes.number.isRequired,
  director: PropTypes.string.isRequired,
  actors: PropTypes.arrayOf(PropTypes.string).isRequired,
}).isRequired;

export {filmProp};
