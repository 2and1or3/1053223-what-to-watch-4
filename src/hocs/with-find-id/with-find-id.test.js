import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import withFindId from './with-find-id.js';
import {filmProp} from '../../props.js';
import NameSpace from '../../reducer/namespace.js';

const mockStore = configureStore();

const MockComponent = (props) => {
  const {currentFilm} = props;

  return (
    <div>
      {JSON.stringify(currentFilm)}
    </div>
  );
};

MockComponent.propTypes = {
  currentFilm: filmProp,
};

const MockComponentWithFindId = withFindId(MockComponent);

it(`Render component with film found by id or default film`, () => {
  const initialState = {
    [NameSpace.DATA]: {
      films: [],
    }
  };
  const store = mockStore(initialState);

  const tree = renderer
    .create(
        <Provider store = {store}>
          <MockComponentWithFindId />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

// npm run test.jest -- hocs/with-find-id/with-find-id.test.js
