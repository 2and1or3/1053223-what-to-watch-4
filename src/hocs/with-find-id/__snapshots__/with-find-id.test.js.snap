// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component with film found by id or default film 1`] = `
<div>
  {"id":"default","title":"default","poster":"default","preview":"default","src":"default","isFavorite":false,"background":"default","backgroundColor":"default","cover":"default","genre":"default","release":"default","description":"default","rating":8.9,"voiceCount":240,"duration":33,"director":"default","actors":["default"]}
</div>
`;
